src/lib/y2partitioner/ui_state.rb

```diff
@@ -51,124 +45,102 @@ def initialize
+    # means default for each widget will be honored).
+    def initialize
+      @statuses = {}
+      @current_status = nil
+      @open_items = {}
+      @overview_tree_pager = nil
     end
 
-    # Title of the bcache section
-    #
-    # @note See note on {.md_raids_label} about why this looks misplaced.
+    # Method to be called when the user operates in a row of a table of devices
+    # or creates a new device.
     #
-    # @return [String]
-    def bcache_label
-      _("Bcache")
-    end
+    # @param device [Y2Storage::Device, Integer] sid or device object
+    def select_row(device)
+      sid = device.respond_to?(:sid) ? device.sid : device
 
-    # Title of the Btrfs section
-    #
-    # @note See note on {.md_raids_label} about why this looks misplaced.
-    #
-    # @return [String]
-    def btrfs_filesystems_label
-      _("Btrfs")
+      current_status&.selected_device = sid
     end
 
-    # @return [Integer, nil] if a row must be selected in a table with devices,
-    #   this returns the sid of the associated device
-    attr_reader :row_sid
-
-    # Method to be called when the user decides to visit a given page by
-    # clicking in one node of the general tree.
+    # Method to be called when the user decides to visit a page by clicking in one node of the
+    # general tree.
     #
     # It remembers the decision so the user is taken back to a sensible point of
     # the tree (very often the last he decided to visit) after redrawing.
     #
-    # @param [CWM::Page] page associated to the tree node
-    def go_to_tree_node(page)
-      self.candidate_nodes =
-        if page.respond_to?(:device)
-          device_page_candidates(page)
-        else
-          [page.label]
-        end
-      # Landing in a new node, so invalidate previous details about position
-      # within a node, they no longer apply
-      self.tab = nil
+    # @note when a page is related to a device, its sid is necessary to compute the
+    # {PageStatus#candidate_pages}. However, we're explicitly avoiding to receive the full page
+    # object here to prevent dependency cycles in the Ruby requires. That's why the page_id can be
+    # an Integer. See also {Widgets::Pages::Page#id}.
+    #
+    # @param page_id [String, Integer] id to identify the page status to track
+    def select_page(page_id)
+      self.current_status = status_for(page_id)
     end
 
     # Method to be called when the user switches to a tab within a tree node.
     #
-    # It remembers the decision so the same tab is showed in case the user stays
-    # in the same node after redrawing.
+    # It remembers the decision so the same tab is showed when the same node will be redraw.
     #
     # @param [CWM::Page, String] page associated to the tab (or just its label)
     def switch_to_tab(page)
-      self.tab = page.respond_to?(:label) ? page.label : page
+      current_status&.active_tab = page.respond_to?(:label) ? page.label : page
```

Comment:
I guess the parameter is the page is, so we can assign it directly.

Reply:
Right. This was a leftover. Fixed, thanks.